### PR TITLE
Implement P2P Network Messaging and Chat Features

### PR DESCRIPTION
**Pull request by scherzma**
This PR introduces significant enhancements to the Skunk project, focusing on peer-to-peer network messaging and chat functionality:

- Implement message handlers for various network operations (sending messages, syncing, joining/leaving chats, inviting peers, sending files, setting usernames)
- Refactor Peer struct to manage network connections, message handlers, and security context
- Introduce MessageSender to handle sending messages over the network
- Create SecurityContext to validate messages based on chat membership and invitations
- Update network and storage port interfaces to support new functionality
- Implement ChatApp service to handle chat-related logic and frontend interactions
- Add SQLite storage adapter for persisting messages, chats, invitations, and other data
- Write unit tests to verify functionality of message handlers, chat logic, and storage operations

Key changes in `cmd/skunk/application/domain/p2p_network/p_service/`, `cmd/skunk/application/port/`, `cmd/skunk/adapter/out/storage/`, and `test/` directories.

This PR lays the foundation for a functional P2P chat application. Future work will focus on UI integration, edge cases, and performance optimization.

Please review and provide feedback.

Related Tickets: #27 #25 #24
Closes: #19

### CODE REVIEW

File: test/inviteToChat_test.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		t.Fatalf("Expected to find invitation to 'chat1' for 'user2', but did not")
+	}
+
+	fmt.Println("Invite to chat test passed")
```
Comment (JavaHammes) [inline test/inviteToChat_test.go:84]: maybe use `t.Log` but you could also just run the test in `verbose mode` using the `-v` flag, which basically does the same thing.

File: test/inviteToChat_test.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+			{PublicKey: "user6.onion", Address: "address2"},
+		},
+	}
+
+	inviteContentBytes, err := json.Marshal(inviteContent)
+	if err != nil {
+		t.Fatalf("Error marshalling invite content: %v", err)
+	}
+
+	//inviteBytes := string(inviteContentBytes)
```
Comment (JavaHammes) [inline test/inviteToChat_test.go:49]: can we delete these comments?

File: test/peer_test.go [hunk 3]
```diff
@@ -2,6 +2,7 @@ package test
```
Comment (JavaHammes) [inline test/peer_test.go:2]: this is not the current version of tests on the main branch. Mabye need to pull?  In theory there shouldn't be any diffs here.

File: test/inviteToChat_test.go (hunk 1 above)
Comment (JavaHammes) [reply test/inviteToChat_test.go:84]: change other tests accordingly please

File: test/tor_test.go [hunk 4]
```diff
@@ -18,7 +18,7 @@ func TestStartTor(t *testing.T) {
 		LocalPort:            "1111",
 		RemotePort:           "2222",
```
Comment (JavaHammes) [inline test/tor_test.go:19]: i think you should also not add these changes and ignore them.

**Review by JavaHammes — CHANGES_REQUESTED** [review]
Looks good! Nice job! Just a couple of short comments.

File: cmd/skunk/adapter/out/storage/storageSQLiteAdapter/StorageSQLiteAdapter.go [hunk 5]
```diff
@@ -0,0 +1,505 @@
...
...sage_id VARCHAR(1024) NOT NULL CONSTRAINT Messages_pk PRIMARY KEY,\n    content TEXT,\n    date INTEGER NOT NULL,\n    operation INTEGER NOT NULL,\n    sender_peer_id VARCHAR(1024) NOT NULL CONSTRAINT Messages_Peers_peer_id_fk REFERENCES Peers ON UPDATE CASCADE,\n    chat_id VARCHAR(1024) NOT NULL CONSTRAINT Messages_Chats_chat_id_fk REFERENCES Chats ON UPDATE CASCADE,\n    receiver_peer_id VARCHAR(1024) NOT NULL CONSTRAINT Messages_Peers_peer_id_fk_2 REFERENCES Peers,\n    sender_address VARCHAR(1024) NOT NULL,\n    receiver_address VARCHAR(1024) NOT NULL\n);\n\nCREATE TABLE IF NOT EXISTS Invitations (\n    invitation_id INTEGER NOT NULL CONSTRAINT Invitations_pk PRIMARY KEY AUTOINCREMENT,\n    invitation_status INTEGER NOT NULL,\n    message_id VARCHAR(1024) NOT NULL CONSTRAINT Invitations_Messages_message_id_fk REFERENCES Messages\n);\n\nCREATE TABLE IF NOT EXISTS PeersInInvitedChat (\n    public_key VARCHAR(1024) NOT NULL,\n    invited_peer_id INTEGER NOT NULL CONSTRAINT PeersInInvitedChat_pk PRIMARY KEY AUTOINCREMENT,\n    address VARCHAR(1024) NOT NULL,\n    invitation_id INTEGER NOT NULL CONSTRAINT PeersInInvitedChat_Invitations_invitation_id_fk REFERENCES Invitations\n);\n"
```
[the remark this answers is not in the document]
Comment (scherzma) [reply cmd/skunk/adapter/out/storage/storageSQLiteAdapter/StorageSQLiteAdapter.go:46]: I agree it will be a good change in the future, this way fixes some problems regarding the path of the SQL file and running tests or the actual program.
Honestly not as clean as i want it to be, but it's pretty stable.

File: cmd/skunk/application/domain/p2p_network/p_service/messageHandlers/syncResponseHandler.go [hunk 6]
```diff
@@ -3,22 +3,27 @@ package messageHandlers
...
 // It retrieves the chat message repository, unmarshals the received messages from the message content,
 // and adds each received message to the chat message repository.
 // TODO: Implement a security check to ensure that the message is valid.
-func (s *SyncResponseHandler) HandleMessage(message network.Message) error {
-
-	chatRepo := p_model.GetNetworkChatsInstance()
-	chatMessageRepo := chatRepo.GetChat(message.ChatID)
+func (s *syncResponseHandler) HandleMessage(message network.Message) error {
+	//chatRepo := p_model.GetNetworkChatsInstance() TODO: change
+	//chatMessageRepo := chatRepo.GetChat(message.ChatID) TODO: change
```
[the remark this answers is not in the document]
Comment (scherzma) [reply cmd/skunk/application/domain/p2p_network/p_service/messageHandlers/syncResponseHandler.go:26]: fixed

File: cmd/skunk/application/domain/p2p_network/p_service/messageHandlers/messageSender.go [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+import (
+	"errors"
+	"fmt"
+	"github.com/scherzma/Skunk/cmd/skunk/application/domain/p2p_network/p_service"
+	"github.com/scherzma/Skunk/cmd/skunk/application/port/network"
+)
+
+// This has many problems right now.
+// For example is it only able to handle on connection. That means, while the peer is able to handle more, this is a bottleneck.
```
[the remark this answers is not in the document]
Comment (scherzma) [reply cmd/skunk/application/domain/p2p_network/p_service/messageHandlers/messageSender.go:11]: done

File: test/inviteToChat_test.go (hunk 2 above)
Comment (scherzma) [reply test/inviteToChat_test.go:49]: done

File: test/inviteToChat_test.go (hunk 1 above)
Comment (scherzma) [reply test/inviteToChat_test.go:84]: done

File: test/peer_test.go (hunk 3 above)
Comment (scherzma) [reply test/peer_test.go:2]: now up do date

File: test/tor_test.go (hunk 4 above)
Comment (scherzma) [reply test/tor_test.go:19]: done
